### ISSUE TITLE
add git-based pull to speed up repo

### ISSUE THREAD

**Issue by shriram** [shriram/gradescope-racket#3]
To avoid Gradescope build times, there's a recommendation (also from Joe Politz) to pull from the autograder from a git repository. 

https://gradescope-autograders.readthedocs.io/en/latest/git_pull/

Then there's only one build, and every update is automatically seen. However, this seems extremely wasteful (a git pull for each student). Furthermore, it's unclear that it's easy to set up for *this* project, because it's meant to be generic: the people customizing it would have to be able to set up their own public git repositories, etc.

The much better solution is to upload an image to Gradescope instead, and let Docker's layer caching, etc., do its job. In the meanwhile, a user comfortable enough with following Gradescope's instructions above can probably figure out how to set this up for themselves.

**Comment by max-heller** [comment]
The way we handled this for the [pyret autograder](https://github.com/brown-cs19/autograder) is by separating the core autograding functionality and assignment specific files into separate repos. The "autograder" uploaded to gradescope is actually the assignment repo, which in its setup clones and sets up the actual autograder, and in its `run_autograder` pulls the latest changes from both repos before running the autograder.

The setup isn't the most elegant, and we have to pull twice for each student, but considering how long pyret takes to compile that time is negligible. We have a private repo for the assignment files, and we include a github deploy key in the zip for gradescope so it can access the assignment.

**Comment by shriram** [comment]
I think I follow most of that. My concern is that this particular repo is meant to be a generic repo for anyone who wants to autograde Racket submissions on Gradescope — it's not specific to my classes, etc. (The intent is that people clone it, then add their own grader.) I worry that some of the educators using it will have difficulty with many of those steps. Wouldn't it be better to just wait for Gradescope handle Docker layers properly?

**Comment by max-heller** [comment]
Yeah, our setup wouldn't work well for that, since it relies on having a template branch of the (private) assignment repo that took some work to set up on its own.
